clash-cores/src/Clash/Cores/Xilinx/Ethernet/Gmii/Internal.hs

```diff
@@ -0,0 +1,400 @@
+            .speed_is_100(0),
+            .speed_is_10_100(0),
+            .signal_detect(~ARG[#{signalDetect}]),
+            .configuration_vector(~ARG[#{pmaConfig}]),
+            .an_adv_config_vector(~ARG[#{pmaAnAdvancedConfig}]),
+            .an_restart_config(~ARG[#{restartAn}]),
+            .rxp(~ARG[#{lvdsInP}]),
+            .rxn(~ARG[#{lvdsInN}]),
+            .gmii_txd(~ARG[#{gmiiTxD}]),
+            .gmii_tx_en(~ARG[#{gmiiTxDv}]),
+            .gmii_tx_er(~ARG[#{gmiiTxEr}]),
+            .clk125_out(~SYM[#{clk125}]),
+            .rst_125_out(~SYM[#{rst125}]),
+            .txp(~SYM[#{lvdsOutP}]),
+            .txn(~SYM[#{lvdsOutN}]),
+            .gmii_rxd(~SYM[#{gmiiRxD}]),
+            .gmii_rx_dv(~SYM[#{gmiiRxDv}]),
+            .gmii_rx_er(~SYM[#{gmiiRxEr}]),
+            .status_vector(~SYM[#{statusVector}])
+          );
+          assign ~RESULT =
+            { ~SYM[#{clk125}]
+            , ~SYM[#{rst125}]
+            , ~SYM[#{lvdsOutP}]
+            , ~SYM[#{lvdsOutN}]
+            , ~SYM[#{gmiiRxD}]
+            , ~SYM[#{gmiiRxDv}]
+            , ~SYM[#{gmiiRxEr}]
+            , ~SYM[#{statusVector}]
+            };
+        includes:
+          - extension: clash.tcl
+            name: gig_ethernet_pcs_pma
+            format: Haskell
+            templateFunction: #{tfName}
+
+          |]
+  )
+  #-}
+
+tclTemplateFunction :: TemplateFunction
+tclTemplateFunction = TemplateFunction used valid tclTemplate
+ where
+  used = [sgmiiClkP .. gmiiTxEr] -- use all arguments
+  valid = const True
+  -- Constraints
+  _knownDomSgmii
+    :< _knownDomGmii
+    :< _domPeriodSgmii
+    :< _domPeriodGmii
+    :< _domEdgeSgmii
+    :< _domEdgeGmii
+    :< _hasAsyncRstSgmii
+    :< _hasSyncRstGmii
+    :< _domRstPolSgmii
+    :< _domRstPolGmii
+    :<
+    -- Arguments
+    sgmiiClkP
+    :< _sgmiiClkN
+    :< _rst
+    :< _signalDetect
+    :< _pmaConfig
+    :< _pmaAnAdvancedConfig
+    :< _restartAn
+    :< _lvdsInP
+    :< _lvdsInN
+    :< _gmiiTxD
+    :< _gmiiTxDv
+    :< gmiiTxEr
+    :< _ =
+      (0...)
+
+tclTemplate :: Backend s => BlackBoxContext -> State s Doc
+tclTemplate bbCtx
+  | [compName] <- bbQsysIncName bbCtx =
+      let
+        bbText =
+          [__di|
+      namespace eval $tclIface {
+        variable api 1
+        variable scriptPurpose createIp
+        variable ipName {#{compName}}
+        proc createIp {ipName0 args} {
+          create_ip -name gig_ethernet_pcs_pma -vendor xilinx.com -library ip \
+            -version 16.2 -module_name $ipName0
+          set_property -dict [list \
+            CONFIG.LvdsRefClk {625} \
+            CONFIG.Standard {SGMII} \
+            CONFIG.Physical_Interface {LVDS} \
+            CONFIG.Management_Interface {false} \
+            CONFIG.Ext_Management_Interface {false} \
+            CONFIG.SGMII_PHY_Mode {false} \
+            CONFIG.SupportLevel {Include_Shared_Logic_in_Core}] [get_ips $ipName0]
+        return
+          }
```

Comment:
These backslashes need to be `\\\&` as I think they were in an earlier version. Currently the generated Tcl probably works but it looks horrendous:
```tcl
  proc createIp {ipName0 args} {
    create_ip -name gig_ethernet_pcs_pma -vendor xilinx.com -library ip             -version 16.2 -module_name $ipName0

    set_property -dict [list             CONFIG.LvdsRefClk {625}             CONFIG.Standard {SGMII}             CONFIG.Physical_Interface {LVDS}             CONFIG.Management_Interface {false}             CONFIG.Ext_Management_Interface {false}             CONFIG.SGMII_PHY_Mode {false}             CONFIG.SupportLevel {Include_Shared_Logic_in_Core}] [get_ips $ipName0]







  return
    }
```